libs/pageserver_api/src/key.rs

```diff
@@ -475,12 +475,14 @@ pub const AUX_FILES_KEY: Key = Key {
 // Reverse mappings for a few Keys.
 // These are needed by WAL redo manager.
 
+pub const NON_INHERITED_RANGE: Range<Key> = AUX_FILES_KEY..AUX_FILES_KEY.next();
+
 // AUX_FILES currently stores only data for logical replication (slots etc), and
 // we don't preserve these on a branch because safekeepers can't follow timeline
 // switch (and generally it likely should be optional), so ignore these.
 #[inline(always)]
 pub fn is_inherited_key(key: Key) -> bool {
-    key != AUX_FILES_KEY
+    !NON_INHERITED_RANGE.contains(&key)
```

Comment:
I assume this would be less efficient than doing a != check but I'm fine with that...

Reply:
Yeah, it's an extra comparison in the worst case, but I wouldn't be surprised if the compiler figures out that: `x <= y < x + 1` is equivalent to `x == y`. I wrote a compiler optimisation which did something similar in the past.